sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/udf/IsNan.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.extensions.sql.impl.udf;
+
+import com.google.auto.service.AutoService;
+
+/**
+ * IS_NAN(X)
+ *
+ * <p>Returns TRUE if the value is a NaN value. Returns NULL for NULL inputs. input: Float, Double
+ *
+ * <p>Output: Boolean
+ */
+@AutoService(BeamBuiltinFunctionClass.class)
+public class IsNan implements BeamBuiltinFunctionClass {
+  private static final String SQL_FUNCTION_NAME = "IS_NAN";
+
+  @UserDefinedFunctionAnnotation(
+    funcName = SQL_FUNCTION_NAME,
+    parameterArray = {Float.class},
+    returnType = Boolean.class
+  )
+  public Boolean isNan(Float value) {
+    return Float.isNaN(value);
+  }
+
+  @UserDefinedFunctionAnnotation(
+    funcName = SQL_FUNCTION_NAME,
+    parameterArray = {Double.class},
```

Comment:
Annotation has [a limited set of types](https://stackoverflow.com/questions/1458535/which-types-can-be-used-for-java-annotation-members).

 I changed types here to `Schema.Typename`.

Reply:
I think it is fine for now but I am not 100% sure. There are different types that have the same beam typename, e.g. TIMESTAMP, DATE, TIME have the same beam type, so when registering it with calcite it can potentially become non-castable